source/mainnet/smart-contracts/general/introduction.rst

```diff
@@ -88,6 +88,22 @@ module <contract-module>`. After this a smart contract can be *initialized* to
 obtain a :ref:`smart contract instance <contract-instances>`. Finally a smart
 contract instance can be repeatedly updated according to its own logic.
 
+.. image:: images/smart-contract-lifecycle.png
+    :width: 100%
+    :alt: flow diagram with different actions
+
+#. In ``cargo-concordium`` :ref:`run the init command<setup-contract>` to start a new project.
+
+#. Edit your contract, including the entrypoints, functions, and parameters necessary to execute what is needed. If :ref:`using a schema<build-schema>`, make sure that the contract is prepared for this.
+
```

Comment:
Maybe it makes sense to include a "Test" box? 
Something like
`Initialize -> Edit -> Test -> Build -> Deploy -> ...`
Or, at least mention it in the description. To emphasize that one can run code off-chain for testing purposes.